modules/apps/frontend-taglib/frontend-taglib-clay/src/main/java/com/liferay/frontend/taglib/clay/servlet/taglib/LinkTag.java

```diff
@@ -0,0 +1,347 @@
+
+import javax.servlet.jsp.JspException;
+import javax.servlet.jsp.JspWriter;
+
+/**
+ * @author Chema Balsas
+ */
+public class LinkTag extends BaseContainerTag {
+
+	@Override
+	public int doStartTag() throws JspException {
+		setAttributeNamespace(_ATTRIBUTE_NAMESPACE);
+
+		setContainerElement("a");
+
+		if (Validator.isNotNull(_ariaLabel)) {
+			setDynamicAttribute(StringPool.BLANK, "aria-label", _ariaLabel);
+		}
+
+		if (Validator.isNotNull(_href)) {
+			setDynamicAttribute(StringPool.BLANK, "href", _href);
+		}
+
+		if (Validator.isNotNull(_target)) {
+			setDynamicAttribute(StringPool.BLANK, "rel", "noreferrer noopener");
+			setDynamicAttribute(StringPool.BLANK, "target", _target);
+		}
+
+		if (Validator.isNotNull(_title)) {
+			setDynamicAttribute(
+				StringPool.BLANK, "title",
+				LanguageUtil.get(
+					TagResourceBundleUtil.getResourceBundle(pageContext),
+					_title));
+		}
+
+		return super.doStartTag();
+	}
+
+	/**
+	 * @deprecated As of Athanasius (7.3.x), with no direct replacement
+	 */
+	@Deprecated
+	public String getAriaLabel() {
+		return _ariaLabel;
+	}
+
+	public boolean getBorderless() {
+		return _borderless;
+	}
+
+	public boolean getButton() {
+		return _button;
+	}
+
+	/**
+	 * @deprecated As of Athanasius (7.3.x), replaced by {@link
+	 *             #getDisplayType()}
+	 */
+	@Deprecated
+	public String getButtonStyle() {
+		return getDisplayType();
+	}
+
+	public String getDisplayType() {
+		return _displayType;
+	}
+
+	public String getDownload() {
+		return _download;
+	}
+
+	public String getHref() {
+		return _href;
+	}
+
+	public String getIcon() {
+		return _icon;
+	}
+
+	/**
+	 * @deprecated As of Athanasius (7.3.x), with no direct replacement
+	 */
+	@Deprecated
+	public String getIconAlignment() {
+		return _iconAlignment;
+	}
+
+	public String getLabel() {
+		return _label;
+	}
+
+	public boolean getMonospaced() {
+		return _monospaced;
+	}
+
+	public boolean getOutline() {
+		return _outline;
+	}
+
+	public boolean getSmall() {
+		return _small;
+	}
+
+	/**
+	 * @deprecated As of Athanasius (7.3.x), replaced by {@link
+	 *             #getDisplayType()}
+	 */
+	@Deprecated
+	public String getStyle() {
+		return getDisplayType();
+	}
+
+	/**
+	 * @deprecated As of Athanasius (7.3.x), with no direct replacement
+	 */
+	@Deprecated
+	public String getTarget() {
+		return _target;
+	}
+
+	/**
+	 * @deprecated As of Athanasius (7.3.x), with no direct replacement
+	 */
+	@Deprecated
+	public String getTitle() {
+		return _title;
+	}
+
+	/**
+	 * @deprecated As of Athanasius (7.3.x), with no direct replacement
+	 */
+	@Deprecated
+	public void setAriaLabel(String ariaLabel) {
+		_ariaLabel = ariaLabel;
+	}
+
+	public void setBorderless(boolean borderless) {
+		_borderless = borderless;
+	}
+
+	public void setButton(boolean button) {
+		_button = button;
+	}
+
+	/**
+	 * @deprecated As of Athanasius (7.3.x), replaced by {@link
+	 *             #setDisplayType(String)}
+	 */
+	@Deprecated
```

Comment:
I'm thinking if we really need to deprecate these java methods. It's an entirely new class/package and for sure no one was using it before from the outside. Maybe the deprecation is only useful on the TLD which is the API surface that will change for people using it.

Reply:
Not sure I follow... what are you suggesting?

If we define the attributes in the tld but they don't have setters, then we will throw a not-found-method exception in the code. If we remove the attribute definition from the tld, then we effectively introduce a breaking change.

Since we want to keep the namespace `clay:link`, this is more or less the balance I found 🤔

Reply:
> If we define the attributes in the tld but they don't have setters, then we will throw a not-found-method exception in the code.

Ah okay. Haven't thought about that.